controllers/provider-openstack/cmd/gardener-extension-provider-openstack/app/app.go

```diff
@@ -82,7 +82,7 @@ func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
 			infraCtrlOpts.Completed().Apply(&openstackinfrastructure.DefaultAddOptions.Controller)
 			infraReconcileOpts.Completed().Apply(&openstackinfrastructure.DefaultAddOptions.IgnoreOperationAnnotation)
 
-			if err := openstackcontroller.AddToManager(mgr); err != nil {
+			if err := openstackcontroller.AddToManager(mgr, mgrOpts.Completed().Disabled); err != nil {
```

Comment:
If you switch the `Disabled` to a standalone `bool` flag you could instead just say
```go
if !disabled {
  openstackcontroller.AddToManager(mgr)
}
```

Reply:
`disabled` should be a map since there are multiple things to disable. I could use something like:

```go
if !disabled[controlplane.ControllerName] {
  awscontrolplane.AddToManager(mgr)
}
```

and so on for all other controllers (and in all `app.go` files), but I don't really like this code - the `AddToManagerBuilder` already provides a useful abstraction that can be used for this, as I did it.

Reply:
I think with the `AddToManagerBuilder` its single responsibility is to accumulate `AddToManager` functions - if it now has this behavior of this 'string map' it's gonna be tightly coupled to command line parsing which it shouldn't since it knows nothing about that.

Reply:
The single responsibility that you see is preserved, just functions are now named which allows filtering. I don't see any tight coupling - the names to filter on are just passed, independently of where they are obtained (command line, env variable, anything else), there is no package level dependency, etc. We have a similar mechanism in SEN and no command line flags whatsoever.

Reply:
Let's maybe look at the signature: Before, the responsibility was clear: It just accumulated all the functions and made it possible to do one `AddToManager` for all of these functions.
Now, it decides on a mapping and names which are not related to anything inside this code - Why should a string to boolean map matter for the `AddToManagerBuilder` whether it should add a controller or not? The only entity that knows to what a name is related is the one that does the flag parsing and thus it should be the one deciding which function to pass into an `AddToManagerBuilder`. In the end, you can probably reuse the `AddToManagerBuilder` but there is another entity that knows how to make the links. I'll provide a quick comment how I mean that.

Reply:
The responsibility is the same and it's as clear as before. Functions are now named so that they can be filtered. The exact origin of the filter is not important (and not being depended upon). It can come from the command line, or from another source. *The only entity that knows to what a name is related is the one that does the flag parsing* is true right now only incidentally because we are using command line flags for configuration in preference to other alternatives. "Filterability" of functions is not accidental, however, it's a "core feature" that we would need even if we don't use command line flags at all.

Reply:
The names and the functions are coupled at a code level. As mentioned, we cannot rename functions anymore without touching core code. That should not happen.
And filterability is definitely one thing you'd like to put on top. The controller used to work perfectly before, you want to add a concern so you add a layer of indirection on top.